src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeConverter.cs

```diff
@@ -364,7 +368,13 @@ protected SimplePropertyDescriptor(Type componentType, string name, Type propert
             /// <summary>
             /// Gets a value indicating whether this property is read-only.
             /// </summary>
-            public override bool IsReadOnly => Attributes.Contains(ReadOnlyAttribute.Yes);
+
+            public override bool IsReadOnly
+            {
+                [DynamicDependency(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor | DynamicallyAccessedMemberTypes.PublicFields, typeof(ReadOnlyAttribute))]
+                [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode", Justification = "The DynamicDependency ensures the correct members are preserved.")]
```

Comment:
Specifically for cases like this and other places where you are suppressing warnings, shouldn't we have a test for that to ensure it is true? I ask since I don't really see Trimming tests added by this PR.

Reply:
A lot of the value we got from writing trimming tests previously was because we didn't have the linker analysis to tell us what is and isn't safe. But we do now, so we are relying heavily on the linker.

I can add a test for this.